src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java

```diff
@@ -493,45 +496,60 @@ private void checkEventTypeBehaviorGroups(Header identityHeader, String eventTyp
         }
     }
 
-    private void checkEndpointHistory(Header identityHeader, String endpointId, int expectedHistoryEntries, boolean expectedInvocationResult, int expectedHttpStatus) {
-        String responseBody = given()
-                .basePath(API_INTEGRATIONS_V_1_0)
-                .header(identityHeader)
-                .pathParam("endpointId", endpointId)
-                .when()
-                .get("/endpoints/{endpointId}/history")
-                .then()
-                .statusCode(200)
-                .contentType(JSON)
-                .extract().body().asString();
+    private void retry(Supplier<Boolean> checkEndpointHistoryResult) {
+        await()
+                .pollInterval(Duration.ofSeconds(1L))
+                .atMost(Duration.ofSeconds(5L))
+                .until(() -> checkEndpointHistoryResult.get());
```

Comment:
Would this fail after the 5 seconds without success?

Reply:
Yes that's the idea

Reply:
The test will try to retrieve the history each second and stop (with an exception) after 5 seconds.